src/Compilers/CSharp/Test/Emit2/Attributes/AttributeTests.cs

```diff
@@ -11291,6 +11291,58 @@ class A<T> : System.Attribute
+  IInvalidOperation (OperationKind.Invalid, Type: null, IsInvalid, IsImplicit) (Syntax: 'Inexistent( ... { _ => 1 })')
+    Children(3):
+        ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: ?) (Syntax: 'SomeProperty = 1')
+          Left:
+            IInvalidOperation (OperationKind.Invalid, Type: ?) (Syntax: 'SomeProperty')
+              Children(0)
+          Right:
+            ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+        ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: ?) (Syntax: 'F = null')
+          Left:
+            IInvalidOperation (OperationKind.Invalid, Type: ?) (Syntax: 'F')
+              Children(0)
+          Right:
+            ILiteralOperation (OperationKind.Literal, Type: null, Constant: null) (Syntax: 'null')
+        ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: ?) (Syntax: 'G = 0 switch { _ => 1 }')
+          Left:
+            IInvalidOperation (OperationKind.Invalid, Type: ?) (Syntax: 'G')
+              Children(0)
+          Right:
+            ISwitchExpressionOperation (1 arms, IsExhaustive: True) (OperationKind.SwitchExpression, Type: System.Int32) (Syntax: '0 switch { _ => 1 }')
+              Value:
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0) (Syntax: '0')
+              Arms(1):
+                  ISwitchExpressionArmOperation (0 locals) (OperationKind.SwitchExpressionArm, Type: null) (Syntax: '_ => 1')
+                    Pattern:
+                      IDiscardPatternOperation (OperationKind.DiscardPattern, Type: null) (Syntax: '_') (InputType: System.Int32, NarrowedType: System.Int32)
+                    Value:
+                      ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 1) (Syntax: '1')
+
+""";
+            var expectedDiagnostics = new[]
+            {
+                // (1,22): error CS0246: The type or namespace name 'InexistentAttribute' could not be found (are you missing a using directive or an assembly reference?)
+                // [assembly: /*<bind>*/Inexistent(SomeProperty = 1, F = null, G = 0 switch { _ => 1 })/*</bind>*/]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Inexistent").WithArguments("InexistentAttribute").WithLocation(1, 22),
+                // (1,22): error CS0246: The type or namespace name 'Inexistent' could not be found (are you missing a using directive or an assembly reference?)
+                // [assembly: /*<bind>*/Inexistent(SomeProperty = 1, F = null, G = 0 switch { _ => 1 })/*</bind>*/]
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "Inexistent").WithArguments("Inexistent").WithLocation(1, 22)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<AttributeSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
```

Comment:
>} [](http://example.com/codeflow?start=8&length=1)

Consider also adding explicit semantic model verification:
```
            var comp = CreateCompilation(source);
            var tree = comp.SyntaxTrees[0];
            var model = comp.GetSemanticModel(tree);
            var one = tree.GetRoot().DescendantNodes().OfType<LiteralExpressionSyntax>().First();

            Assert.Equal("1", one.ToString());
            Assert.Equal("SomeProperty = 1", one.Parent.ToString());
            Assert.Equal("System.Int32", model.GetTypeInfo(one).Type.ToTestDisplayString());
```

Reply:
I don't think the semantic model validation adds much on top of IOperation check, which already shows how everything got bound.